test/Polly.RateLimiting.Tests/RateLimiterResilienceStrategyTests.cs

```diff
@@ -81,29 +83,33 @@ public async Task Execute_LeaseRejected(bool hasEvents, bool hasRetryAfter)
 
         outcome.Exception!.StackTrace.Should().Contain("Execute_LeaseRejected");
 
-        _limiter.VerifyAll();
-        _lease.VerifyAll();
         eventCalled.Should().Be(hasEvents);
 
-        _diagnosticSource.VerifyAll();
+        await _limiter.ReceivedWithAnyArgs().AcquireAsync(default, default);
+        _lease.Received().Dispose();
     }
 
-    private void SetupLimiter(CancellationToken token) => _limiter
-                .Protected()
-                .Setup<ValueTask<RateLimitLease>>("AcquireAsyncCore", 1, token)
-                .Returns(new ValueTask<RateLimitLease>(_lease.Object));
+    private void SetupLimiter(CancellationToken token)
+    {
+        var result = new ValueTask<RateLimitLease>(_lease);
+        _limiter
+            .GetType()
+            .GetMethod("AcquireAsyncCore", BindingFlags.NonPublic | BindingFlags.Instance)!
```

Comment:
:(

Was not nice even before though ...

Reply:
Yeah, protected mocking always looks ugly.